app/Http/Controllers/ChirpController.php

```diff
@@ -95,13 +97,17 @@ public function destroy(Chirp $chirp): RedirectResponse
      * display the chirps from only the past week
      */
 
-     public function recent()
+    public function recent(Request $request, $days = 14)
      {
-         $days = 14; // Default to 14 days 
-         $startDate = Carbon::now()->subDays($days);
-         $chirps = Chirp::where('created_at', '>=', $startDate)->get();
+        $request->validate([
+            'days' => 'nullable|numeric|min:1'
```

Comment:
This rule would also allow decimals like 1.5 but it doesn't break it so keep it as is. Another validation rule that could have been used here is the regex rule which is very powerful and a good validator to keep in mind when you add validation for other future complex inputs. Good job!